website/content/en/docs/ansible/dev/webhooks.md

```diff
@@ -0,0 +1,166 @@
+
+## Deploying the webhook server
+
+To deploy the webhook server as a sidecar alongside your operator, all you need to do is add the container
+specification to your `deploy/operator.yaml`. You may also need to add a volume for mounting in TLS secrets,
+as your webhook server is required to have a valid SSL configuration. Below is a sample updated container
+specification that deploys a webhook:
+
+```yaml
+      containers:
+        - name: my-operator
+          # Replace this with the built image name
+          image: "REPLACE_IMAGE"
+          imagePullPolicy: "Always"
+          volumeMounts:
+          - mountPath: /tmp/ansible-operator/runner
+            name: runner
+          env:
+            - name: WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: OPERATOR_NAME
+              value: "validating-operator"
+            - name: ANSIBLE_GATHERING
+              value: explicit
+        # This deploys the webhook
+        - name: webhook
+          # Replace this with the built image name
+          image: "REPLACE_WEBHOOK_IMAGE"
+          imagePullPolicy: "Always"
+          volumeMounts:
+          - mountPath: /etc/tls/
+            name: webhook-cert
+      volumes:
+        - name: runner
+          emptyDir: {}
+        # This assumes there is a secret called webhook-cert containing TLS certificates
+        # Projects like cert-manager can create these certificates
+        - name: webhook-cert
+          secret:
+            secretName: webhook-cert
+```
+
+This will run your webhook server alongside the operator, but Kubernetes will not yet call the webhooks before
+resources can be created. In order to let Kubernetes know about your webhooks, you must create specific API resources.
+
+## Making Kubernetes call your webhooks
+
+In order to make your webhooks callable at all, first you must create a `Service` that points at your
+webhook server. Below is a sample service that creates a `Service` named `my-operator-webhook`, that will
+send traffic on port `443` to port `5000` in a `Pod` that matches the selector `name=my-operator`. Modify these
+values to match your environment.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-operator-webhook
+spec:
+  ports:
+  - name: webhook
+    port: 443
+    protocol: TCP
+    # Change targetPort to match the port your server is listening on
+    targetPort: 5000
+  selector:
+    # Change this selector to match the labels on your operator pod
+    name: my-operator
+  type: ClusterIP
+```
+
+Now that you have a `Service` directing traffic to your webhook server, you will need to create
+`MutatingWebhookConfiguration` or `ValidatingWebhookConfiguration` objects (depending on what type of webhook you have deployed), which will tell Kubernetes
```

Comment:
Could we add the links for the refs in k8s docs over `MutatingWebhookConfiguration` or `ValidatingWebhookConfiguration`?WDYT?